tree/tree.go

```diff
@@ -0,0 +1,656 @@
+		child := children.At(i)
+		setDepths(child.(*Node), depth+1)
+	}
+}
+
+// setYOffsets updates each Node's yOffset based on how many items are "above" it
+func setYOffsets(t *Node) {
+	children := t.tree.Children()
+	above := 0
+	for i := 0; i < children.Length(); i++ {
+		child := children.At(i)
+		if child, ok := child.(*Node); ok {
+			child.yOffset = t.yOffset + above + i + 1
+			setYOffsets(child)
+			above += child.Size() - 1
+		}
+	}
+}
+
+// YOffset returns the vertical offset of the selected node.
+// Useful for scrolling to the selected node using a viewport.
+func (m *Model) YOffset() int {
+	return m.yOffset
+}
+
+// Node returns the item at the given yoffset
+func (m *Model) Node(yoffset int) *Node {
+	return findNode(m.root, yoffset)
+}
+
+// NodeAtCurrentOffset returns the item at the current yoffset
+func (m *Model) NodeAtCurrentOffset() *Node {
+	return findNode(m.root, m.yOffset)
+}
+
+// Enumerator sets the enumerator for the tree
+func (m *Model) Enumerator(enumerator ltree.Enumerator) *Model {
+	m.enumerator = &enumerator
+	m.root.Enumerator(enumerator)
+	return m
+}
+
+// Indenter sets the indenter for the tree
+func (m *Model) Indenter(indenter ltree.Indenter) *Model {
+	m.indenter = &indenter
+	m.root.Indenter(indenter)
+	return m
+}
+
+// Since the selected node changes, we need to capture m.yOffset in the
+// style function's closure again
+func (m *Model) updateStyles() {
+	if m.root != nil {
+		m.root.RootStyle(m.rootStyle())
+	}
+
+	items := m.AllNodes()
+	opts := m.getItemOpts()
+	for _, item := range items {
+		item.opts = *opts
+	}
+}
+
+func (m *Model) getItemOpts() *itemOptions {
+	return &itemOptions{
+		openCharacter:   m.OpenCharacter,
+		closedCharacter: m.ClosedCharacter,
+		treeYOffset:     m.yOffset,
+		styles:          m.styles,
+	}
+}
+
+// // selectedNodeStyle sets the node style
+// // and takes into account whether it's selected or not
+// func (m *Model) selectedNodeStyle() StyleFunc {
+// 	return func(children Nodes, i int) lipgloss.Style {
+// 		child := children.At(i)
+// 		if child.yOffset == m.yOffset {
+// 			return m.styles.selectedNodeFunc(children, i)
+// 		}
+//
+// 		return m.styles.nodeFunc(children, i)
+// 	}
+// }
+
+func (m *Model) rootStyle() lipgloss.Style {
+	if m.root.yOffset == m.yOffset {
+		return m.styles.selectedNodeFunc(Nodes{m.root}, 0)
+	}
+
+	return m.styles.rootNodeFunc(Nodes{m.root}, 0)
+}
+
+// findNode starts a DFS search for the node with the given yOffset
+// starting from the given item
+func findNode(t *Node, yOffset int) *Node {
+	if t.yOffset == yOffset {
+		return t
+	}
+
+	children := t.tree.Children()
+	for i := 0; i < children.Length(); i++ {
+		child := children.At(i)
+		if child, ok := child.(*Node); ok {
+			found := findNode(child, yOffset)
+			if found != nil {
+				return found
+			}
+		}
+	}
+
+	return nil
+}
+
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+func max(a, b int) int {
+	if a > b {
+		return a
+	}
+	return b
+}
```

Comment:
Those are not needed anymore, as we recently upgraded the Go version.

`min` and `max` are not generic Go builtins.